st2common/st2common/services/sensor_watcher.py

```diff
@@ -95,9 +95,18 @@ def start(self):
             self.connection.release()
 
     def stop(self):
+        LOG.debug('Shutting down sensor watcher.')
         try:
             if self._updates_thread:
                 self._updates_thread = eventlet.kill(self._updates_thread)
+
+            if self.connection:
+                channel = self.connection.channel()
```

Comment:
Yeah, I'm all for explicitly deleting it (normal scenario), but I was also talking about the edge case scenario where we can't ensure this code will run - e.g. if process crashes or isn't shut down cleanly.

In that case this code won't run so queue will stay there. We can avoid that by using a non durable and auto deletable queue - in this case queue will be deleted when connection to the broken is lost.